src/main/java/com/arrl/radiocraft/common/radio/BandReloadListener.java

```diff
@@ -9,40 +9,44 @@
 import net.minecraft.server.packs.resources.SimpleJsonResourceReloadListener;
 import net.minecraft.util.profiling.ProfilerFiller;
+import org.jetbrains.annotations.NotNull;
 
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 public class BandReloadListener extends SimpleJsonResourceReloadListener {
 
 	private static final Gson GSON = new Gson();
-	private final Map<Integer, Band> bands = new HashMap<>();
+	private final Map<Integer, Band> bands = Collections.synchronizedMap(new HashMap<>());
 
 	public BandReloadListener(String directory) {
 		super(GSON, directory);
 	}
 
 	@Override
-	protected void apply(Map<ResourceLocation, JsonElement> jsonMap, ResourceManager resourceManager, ProfilerFiller profiler) {
-		bands.clear();
-		jsonMap.forEach((resourceLocation, jsonElement) -> {
-			try {
-				JsonObject jsonObject = jsonElement.getAsJsonObject();
-				int wavelength = jsonObject.get("wavelength").getAsInt();
-				int losRange = jsonObject.get("los").getAsInt();
-				int minSkipDay = jsonObject.get("minSkipDay").getAsInt();
-				int maxSkipDay = jsonObject.get("maxSkipDay").getAsInt();
-				int minSkipNight = jsonObject.get("minSkipNight").getAsInt();
-				int maxSkipNight = jsonObject.get("maxSkipNight").getAsInt();
-				int minFrequency = jsonObject.get("minFrequency").getAsInt();
-				int maxFrequency = jsonObject.get("maxFrequency").getAsInt();
-
-				bands.put(wavelength, new Band(wavelength, losRange, minSkipDay, maxSkipDay, minSkipNight, maxSkipNight, minFrequency, maxFrequency));
-
-			} catch (IllegalArgumentException | JsonParseException jsonparseexception) {
-				Radiocraft.LOGGER.error("Parsing error radio band {}: {}", resourceLocation, jsonparseexception.getMessage());
-			}
-		});
+	protected void apply(Map<ResourceLocation, JsonElement> jsonMap, @NotNull ResourceManager resourceManager, @NotNull ProfilerFiller profiler) {
+		synchronized (bands) {
+			bands.clear();
+			jsonMap.forEach((resourceLocation, jsonElement) -> {
+				try {
+					JsonObject jsonObject = jsonElement.getAsJsonObject();
+					int wavelength = jsonObject.get("wavelength").getAsInt();
+					int losRange = jsonObject.get("los").getAsInt();
+					int minSkipDay = jsonObject.get("minSkipDay").getAsInt();
+					int maxSkipDay = jsonObject.get("maxSkipDay").getAsInt();
+					int minSkipNight = jsonObject.get("minSkipNight").getAsInt();
+					int maxSkipNight = jsonObject.get("maxSkipNight").getAsInt();
+					int minFrequency = jsonObject.get("minFrequency").getAsInt();
+					int maxFrequency = jsonObject.get("maxFrequency").getAsInt();
+
+					bands.put(wavelength, new Band(wavelength, losRange, minSkipDay, maxSkipDay, minSkipNight, maxSkipNight, minFrequency, maxFrequency));
+
+				} catch (IllegalArgumentException | JsonParseException jsonparseexception) {
+					Radiocraft.LOGGER.error("Parsing error radio band {}: {}", resourceLocation, jsonparseexception.getMessage());
+				}
+			});
+		}
```

Comment:
Why do we even allow reloading this? I'd wager this should just be part of data gen and that we just make it so that if a data pack writes it we read it once on game load? The only thing I can think of is if this is synced to the client for use in audio postporcessing to attenuate signals or add noise (but the server can just remind it or something because the server has to tell it the actual value anyway)

Reply:
(Ironically, it's not contained, 2m is `149.896229MHz`)

Reply:
Hmm, okay. I think we should have a separate conversation around whether exposing this to data packs even makes sense (because if these values are missing, particular radios might just crash instead)